pkg/ipcache/metadata.go

```diff
@@ -274,16 +305,18 @@ func FilterMetadataByLabels(filter labels.Labels) []string {
 	return matching
 }
 
-// RemoveLabelsFromIPs wraps RemoveLabels to provide a convenient method for
-// the caller to remove all given prefixes at once. This function will trigger
-// policy update and recalculation if necessary on behalf of the caller if any
-// changes to the kube-apiserver were detected.
+// removeLabelsFromIPs provides a convenient method for the caller to remove
+// all given prefixes at once. This function should be the inverse of
+// InjectLabels(). This function will trigger policy update and recalculation
+// if necessary on behalf of the caller.
 //
-// Identities allocated by InjectLabels() may be released by RemoveLabels().
+// Identities allocated by InjectLabels() may be released by removeLabels().
```

Comment:
Good point. I've clarified this with the following diff. I removed the mention to `may` as it doesn't really help, but opted to simply state that `RemoveLabelsExcluded` will do the inverse of `InjectLabels`, so that it's clearer what can happen in terms of identities at a high-level. Other godocs on the unexported `remove*()` functions should provide more insight as to when an identity is released.

```diff
commit 760ee1445ecd5e89c1340dd0aa0efebb61e8d5b8
Author: Chris Tarazi <chris@isovalent.com>
Date:   Fri Dec 17 12:03:00 2021 -0800

    fixup! ipcache: Generate new identity if label is removed
    
    Signed-off-by: Chris Tarazi <chris@isovalent.com>

diff --git a/pkg/ipcache/metadata.go b/pkg/ipcache/metadata.go
index 22d2415423..1e2d70b49a 100644
--- a/pkg/ipcache/metadata.go
+++ b/pkg/ipcache/metadata.go
@@ -73,12 +73,15 @@ func GetIDMetadataByIP(prefix string) labels.Labels {
 // InjectLabels injects labels from the identityMetadata (IDMD) map into the
 // identities used for the prefixes in the IPCache. The given source is the
 // source of the caller, as inserting into the IPCache requires knowing where
-// this updated information comes from.
+// this updated information comes from. Conversely, RemoveLabelsExcluded
+// performs the inverse: removes labels from the IDMD map and releases
+// identities allocated by this function.
 //
 // Note that as this function iterates through the IDMD, if it detects a change
 // in labels for a given prefix, then this might allocate a new identity. If a
 // prefix was previously associated with an identity, it will get deallocated,
-// so a balance is kept.
+// so a balance is kept, ensuring a one-to-one mapping between prefix and
+// identity.
 func InjectLabels(src source.Source, updater identityUpdater, triggerer policyTriggerer) error {
 	if IdentityAllocator == nil || !IdentityAllocator.IsLocalIdentityAllocatorInitialized() {
 		return ErrLocalIdentityAllocatorUninitialized
@@ -305,12 +308,9 @@ func filterMetadataByLabels(filter labels.Labels) []string {
 	return matching
 }
 
-// removeLabelsFromIPs provides a convenient method for the caller to remove
-// all given prefixes at once. This function should be the inverse of
-// InjectLabels(). This function will trigger policy update and recalculation
-// if necessary on behalf of the caller.
-//
-// Identities allocated by InjectLabels() may be released by removeLabels().
+// removeLabelsFromIPs removes all given prefixes at once. This function should
+// be the inverse of InjectLabels(). This function will trigger policy update
+// and recalculation if necessary on behalf of the caller.
 //
 // A prefix will only be removed from the IDMD if the set of labels becomes
 // empty.
```